### PR TITLE
Update docs to Data Prepper 1.4

### PR DESCRIPTION
**Pull request by keithhc2**
Update Data Prepper docs to Data Prepper 1.4

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _clients/data-prepper/data-prepper-reference.md [hunk 1]
```diff
@@ -65,7 +68,30 @@ request_timeout | No | Integer | The request timeout in millis. Default is `10_0
...
+health_check_service | No | Boolean | Enables a gRPC health check service under `grpc.health.v1/Health/Check`. Default is `false`.
+proto_reflection_service | No | Boolean | Enables a reflection service for Protobuf services (see [gRPC reflection](https://github.com/grpc/grpc/blob/master/doc/server-reflection.md) and [gRPC Server Reflection Tutorial](https://github.com/grpc/grpc-java/blob/master/documentation/server-reflection-tutorial.md) docs). Default is `false`.
+unframed_requests | No | Boolean | Enable requests not framed using the gRPC wire protocol.
+thread_count | No | Integer | The number of threads to keep in the ScheduledThreadPool. Default is `200`.
+max_connection_count | No | Integer | The maximum allowed number of open connections. Default is `500`.
+ssl | No | Boolean | Enables connections to the OTel source port over TLS/SSL. Defaults to `true`.
+sslKeyCertChainFile | Conditionally | String | File-system path or AWS S3 path to the security certificate (e.g. `"config/demo-data-prepper.crt"` or `"s3://my-secrets-bucket/demo-data-prepper.crt"`). Required if ssl is set to `true`.
```
Comment (lizsnyder) [inline _clients/data-prepper/data-prepper-reference.md:87]: `ssl`

File: _clients/data-prepper/data-prepper-reference.md [hunk 2]
```diff
@@ -65,7 +68,30 @@ request_timeout | No | Integer | The request timeout in millis. Default is `10_0
...
+sslKeyFile | Conditionally | String | File-system path or AWS S3 path to the security key (e.g. `"config/demo-data-prepper.key"` or `"s3://my-secrets-bucket/demo-data-prepper.key"`). Required if ssl is set to `true`.
+useAcmCertForSSL | No | Boolean | Whether to enable TLS/SSL using certificate and private key from AWS Certificate Manager (ACM). Default is `false`.
+acmCertificateArn | Conditionally | String | Represents the ACM certificate ARN. ACM certificate take preference over S3 or local file system certificate. Required if `useAcmCertForSSL` is set to `true`.
+awsRegion | Conditionally | String | Represents the AWS region to use ACM or S3. Required if `useAcmCertForSSL` is set to `true` or `sslKeyCertChainFile` and `sslKeyFile` are AWS S3 paths.
```
Comment (lizsnyder) [inline _clients/data-prepper/data-prepper-reference.md:91]: We capitalize "Region"

File: _clients/data-prepper/pipelines.md [hunk 3]
```diff
@@ -115,6 +119,91 @@ service-map-pipeline:
...
+        trace_analytics_service_map: true
+```
+
+Note that it is recommended to scale the `buffer_size` and `batch_size` by the estimated maximum batch size in the client request payload to maintain similar ingestion throughput and latency as in [Classic](#classic).
+
+### Metrics pipeline
+
+Data Prepper supports metrics ingestion using OTel. It currently supports the following metric types:
+
+* Guage
```
Comment (lizsnyder) [inline _clients/data-prepper/pipelines.md:185]: Gauge?

**Review by lizsnyder — APPROVED** [review]

File: _clients/data-prepper/pipelines.md (hunk 3 above)
Comment (dlvenable) [reply _clients/data-prepper/pipelines.md:185]: That was a typo on my part. It should be gauge.

Comment (keithhc2) [reply _clients/data-prepper/pipelines.md:185]: Fixed

**Review by dlvenable — APPROVED** [review]

**Review by alicejw-aws — APPROVED** [review]
LGTM

**Pull request merged**